frontend/src/components/ProductList.tsx

```diff
@@ -0,0 +1,86 @@
+import ProductDetail from "./ProductDetail";
+import { Grid, Paper, Button } from "@material-ui/core";
+import { useProductListStyles } from "../styles/ProductListStyles";
+import { useHistory } from "react-router-dom";
+
+let data = [
+  {
+    name: "product1",
+    category: "category1",
+    price: 15,
+    id: 1,
+  },
+  {
+    name: "product2",
+    category: "category1",
+    price: 30,
+    id: 2,
+  },
+  {
+    name: "product3",
+    category: "category2",
+    price: 10,
+    id: 3,
+  },
+];
+
+function ProductList() {
+  const classes = useProductListStyles();
+  let history = useHistory();
+  let products = data.map((product, index) => (
+    <Grid container key={index}>
+      <Grid item className={classes.detail}>
+        <ProductDetail {...product} />
```

Comment:
Nice use of the `...` separator operator. Another option would have been to just pass the entire product but this way is good as well.